private/functions/Invoke-ManagedComputerCommand.ps1

```diff
@@ -90,68 +90,37 @@ function Invoke-ManagedComputerCommand {
-                # lol I'm not sure how to catch the last error so...
-                try {
-                    $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 15
-                    $parms.ArgumentList = $ArgumentList
-                    Invoke-Command2 @parms -ComputerName $hostname
-                } catch {
-                    try {
-                        $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 14
-                        $parms.ArgumentList = $ArgumentList
-                        Invoke-Command2 @parms -ComputerName $hostname
-                    } catch {
-                        try {
-                            $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 13
-                            $parms.ArgumentList = $ArgumentList
-                            Invoke-Command2 @parms -ComputerName $hostname
-                        } catch {
-                            try {
-                                $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 12
-                                $parms.ArgumentList = $ArgumentList
-                                Invoke-Command2 @parms -ComputerName $hostname
-                            } catch {
-                                try {
-                                    $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 11
-                                    $parms.ArgumentList = $ArgumentList
-                                    Invoke-Command2 @parms -ComputerName $hostname
-                                } catch {
-                                    try {
-                                        $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 10
-                                        $parms.ArgumentList = $ArgumentList
-                                        Invoke-Command2 @parms -ComputerName $hostname
-                                    } catch {
-                                        try {
-                                            $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 9
-                                            $parms.ArgumentList = $ArgumentList
-                                            Invoke-Command2 @parms -ComputerName $hostname
-                                        } catch {
-                                            $ArgumentList[$ArgumentList.GetUpperBound(0)].version = 8
-                                            $parms.ArgumentList = $ArgumentList
-                                            Invoke-Command2 @parms -ComputerName $hostname
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
+                # Log the failure and proceed to the next version
+                Write-Message -Level Verbose -Message "Local connection attempt to $computer failed | $PSItem. Connecting remotely (Version $version)."
```

Comment:
Here, not the local connection attempt failed, but the remote connection attempt with the specific version.

Reply:
The local connection has already failed; otherwise, we wouldn't find ourselves in this catch block where we're attempting the same operation with different versions. I share your concern about the error message being potentially misleading. Do you have any suggestions for a clearer alternative approach?

Reply:
the newest commit should be more specific!